crates/bevy_reflect/bevy_reflect_derive/src/field_attributes.rs

```diff
@@ -117,8 +116,13 @@ fn parse_meta(args: &mut ReflectFieldAttr, meta: &Meta) -> Result<(), syn::Error
             format!("unknown attribute parameter: {}", path.to_token_stream()),
         )),
         Meta::NameValue(pair) if pair.path.is_ident(DEFAULT_ATTR) => {
-            let lit = &pair.lit;
-            match lit {
+            let Expr::Lit(lit) = &pair.value else {
+                Err(syn::Error::new(
+                    pair.span(),
+                    format!("expected a string literal containing the name of a function, but found: {}", pair.to_token_stream()),
+                ))?
+            };
```

Comment:
Are we trying to maintain the same behavior with this PR? If not, I would love to have the `meta` parse this directly as an `ExprPath` rather than having to parse a string literal and then parse that into the `ExprPath`.

If not, no worries, I can make a followup PR!

Reply:
I'm ok with a followup PR 😄